protocol/contracts/TrucoMatch.sol

```diff
@@ -10,42 +10,131 @@ import "./crypt/DeckCrypt.sol";
 
 contract TrucoMatch  {
 
+    struct player {
+        address playerAddress;
+        uint256 tokensAtStake;
+    }
+    
     struct Match {
-        address[2] players; // player 0 is the creator of the match
+        player[2] players; // player 0 is the creator of the match
         CardsStructs.Deck deck;
         CardsStructs.GameState gameState;
         uint256 tokensAtStake; 
+
     }
 
     IERC3333 trucoEngine;
     DeckCrypt deckCrypt;
-    IERC20 trucoin;
+    IERC20 truCoin;
     Match currentMatch;
+    bool isDealOpen;
+
+    // Events
+    event MatchCreated(address match_address);
+    event MatchStarted(address player1, address player2, uint256 tokensAtStake);
+    event PlayerStaked(address player, uint256 tokensAtStake);
+    event DealStarted(address shuffler);
+    event DealEnded();
+
+
+    constructor (IERC3333 _trucoEngine, DeckCrypt _deckCrypt, IERC20 _truCoin, uint256 _tokensAtStake) {
 
-    constructor (IERC3333 _trucoEngine, DeckCrypt _deckCrypt, IERC20 _trucoin) {
         trucoEngine = _trucoEngine;
         deckCrypt = _deckCrypt;
-        currentMatch.players[0] = msg.sender;
+        truCoin = _truCoin;
+        currentMatch.tokensAtStake = _tokensAtStake;
+        currentMatch.players[0].playerAddress = msg.sender;
+
+        //Mint Sould Bound Token
 
-        //Transfer Trucoins from owner to contract
-        //Mint Sould Bound Token 
+        emit MatchCreated(address(this));
```

Comment:
At this point `Engine.startGame` should be called, where it deducts fees and returns the initial game state to work with